src/namespace.lisp

```diff
@@ -53,6 +55,18 @@
       (error "Cannot provide ERRORP-ARG-IN-ACCESSOR-P when CONDITION-NAME ~
               is null."))))
 
+(defun check-namespace-definer-spec (definer)
+  (or (symbolp definer)
+      (when (listp definer)
+        (case (first definer)
+          (function t)
+          (quote (and (symbolp (second definer))
+                      (not (cddr definer))))
+          (lambda (listp (second definer)))
+          (t (listp (first definer)))))
+      (error "Malformed definer ~S"
+             definer)))
```

Comment:
I'd rather express this in terms of the type system for some clarity:

```suggestion
(defun check-namespace-definer-spec (definer)
  (assert (typep definer '(or
                           symbol
                           (cons (eql function))
                           (cons (eql quote) (cons symbol null))
                           (cons (eql lambda) (cons list))
                           (cons list)))
          () "Malformed definer ~S" definer))
```

Please check if my suggestion is equivalent.